lib/storage/azure.js

```diff
@@ -170,33 +171,65 @@ module.exports = function() {
           if (err) {
             return cb(err);
           }
+
           blobSvcs.push({
             svc: svc,
             container: cluster.container || options.container
           });
+
           return cb();
         });
       }, callback);
     },
 
-    copyIn: function(localPath, path, options, callback) {
-      // TODO be more selective about when we do this
-      gzipme(localPath, false, 'best', function() {
-        path = path[0] === '/' ? path.slice(1) : path;
-        async.each(blobSvcs, function(blobSvc, cb) {
-          createContainerBlob(blobSvc, path, localPath, function(err) {
-            if (err) {
-              __log('! Cannot create blob in container : ' + err);
-            }
-            cb(err);
+    copyIn: function(localPath, _path, options, callback) {
+      const path = _path[0] === '/' ? _path.slice(1) : _path;
+      const tmpFileName = Math.random().toString(36).substring(7);
+
+      // store gzipped file in temp dir
+      fs.mkdtemp(this.options.tempPath, (e, dir) => {
+        if (e) {
+          console.log(e);
+          return callback(e);
+        } else {
+          const tempPath = dir + '/' + tmpFileName;
+          let inp = fs.createReadStream(localPath);
+          let out = fs.createWriteStream(tempPath);
+
+          inp.on('error', e => {
+            console.log("Error in write stream", e);
```

Comment:
Should be "error in read stream" yes?

Also, as I imagine neither of us is quite sure whether one or both streams will fire an error, use a flag variable to track whether an error has been reported and discard any extra one.

And, we need to clean up properly on errors. Close the streams, delete the file and the folder if and only if they exist.